File: client/python/mesh/system/mesh_publisher.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import List

from mesh.kinds import Event
from mesh.macro import spi
from mesh.mpc import ServiceProxy
from mesh.prsim import Publisher


@spi("mesh")
class MeshPublisher(Publisher):

    def __init__(self):
        self.proxy = ServiceProxy.default_proxy(Publisher)

    def publish(self, events: List[Event]) -> List[str]:
        return self.proxy.publish(events)

    def broadcast(self, events: List[Event]) -> List[str]:
        return self.proxy.broadcast(events)

File: client/java/src/main/java/io/be/mesh/system/MeshPublisher.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.system;

import io.be.mesh.macro.SPI;
import io.be.mesh.mpc.ServiceProxy;
import io.be.mesh.prsim.Publisher;
import io.be.mesh.struct.Event;

import java.util.List;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshPublisher implements Publisher {

    private final Publisher publisher = ServiceProxy.proxy(Publisher.class);

    @Override
    public List<String> publish(List<Event> event) {
        return publisher.publish(event);
    }

    @Override
    public List<String> broadcast(List<Event> event) {
        return publisher.broadcast(event);
    }
}
